autoload/textobj/sentence.vim

```diff
@@ -87,7 +91,9 @@ function! textobj#sentence#init(...)
 
   " matching against end of sentence, '!', '?', and non-abbrev '.'
   let l:re_term =
-      \ '([!?]|(' .
+      \ '([!?]' .
+      \ g:re_extra_sentence_term .
```

Comment:
A cursory look at your question ... wouldn't the extra expressions you are trying to inject need to go *inside* the character class expression rather than after it? Something like:

```suggestion
      \ '([!?'. g:re_extra_sentence_term .']' .
```

Reply:
Oh hmm you aren't adding characters you're adding whole expressions. We probably need to setup all our matches as an array of expressions and join them at the last minute using `(one|two|three)` option grouping instead of a character class.

Reply:
> Oh hmm you aren't adding characters you're adding whole expressions. We probably need to setup all our matches as an array of expressions and join them at the last minute using `(one|two|three)` option grouping instead of a character class.

Yes, I want to allow whole expressions.  Let me give an example and demonstrate the problem:
```tex
\documentclass{minimal}
\begin{document}

\newcommand{\beq}{\begin{equation}}
\newcommand{\eeq}{\end{equation}}

This is a long sentence, exceeding eighty characters and containing an equation environment below, starting here:
\beq
  x = 1 \,,
\eeq
and it is followed by another long sentence that exceeds eighty characters and should be wrapped by \texttt{gqis}.

\end{document}
```
Consider this with
```vim
let g:re_extra_sentence_term =
  \ '|\\beq' .
  \ '|\\eeq'
  
call textobj#sentence#init()
```
With my proposed implementation, when I place the cursor on the line before the equation and hit `gqis`, it does (almost) the right thing:
```
This is a long sentence, exceeding eighty characters and containing an equation
environment below, starting here: \beq
...
```
(I noticed only now it's not perfect, the `\beq` should actually stay on its own line ideally, but that may be a paragraph issue rather than a sentence issue).

Setting that aside, the main trouble is that hitting `gqis` when the cursor is in the line *after* the `\eeq`, it does not rewrap that but rather the cursor jumps to above the equation and gives again
```
This is a long sentence, exceeding eighty characters and containing an equation
environment below, starting here: \beq
...
```
As far as I can tell, the reason is that the negative lookback jumps over the `\eeq` and `\beq` somehow.